src/plugins/discover/public/application/main/state_management/discover_data_state_container.ts

```diff
@@ -185,6 +185,12 @@ export function getDataStateContainer({
     documents$: new BehaviorSubject<DataDocumentsMsg>(initialState),
     totalHits$: new BehaviorSubject<DataTotalHitsMsg>(initialState),
   };
+  // This is debugging code, helping you to understand which messages are sent to the data observables
+  // Adding a debugger in the functions can be helpful to understand what triggers a message
+  // dataSubjects.main$.subscribe((msg) => addLog('dataSubjects.main$', msg));
+  // dataSubjects.documents$.subscribe((msg) => addLog('dataSubjects.documents$', msg));
+  // dataSubjects.totalHits$.subscribe((msg) => addLog('dataSubjects.totalHits$', msg););
+  // Add window.ELASTIC_DISCOVER_LOGGER = 'debug' to see messages in console
```

Comment:
uncommenting the code can be very helpful to unterstand data fetching observables race conditions, I think it's worth keeping it here

Reply:
This could be helpful to debug at least a part of the staff you mention in https://github.com/elastic/kibana/issues/165192#issuecomment-2085312808 @jughosta